bin/main.rb

```diff
@@ -1,2 +1,46 @@
 #!/usr/bin/env ruby
-puts 'Hello world!'
+require_relative '../lib/player'
+require_relative '../lib/game'
+require_relative '../lib/helper'
+
+puts 'Please enter player1 name: '
+player_1_name = gets.chomp
+player_1_name = Helper.character_only(player_1_name)
+player_1_name = Helper.validate_player_name(player_1_name)
+player1 = Player.new(player_1_name, 'x')
+
+puts 'Please enter player2 name: '
+player_2_name = gets.chomp
+player_2_name = Helper.character_only(player_2_name)
+player_2_name = Helper.validate_player_name(player_1_name, player_2_name)
+player2 = Player.new(player_2_name, 'o')
+
+game = Game.new(player1, player2)
+puts '- - - - - - - -'
+puts "Welcome to the Tic - Tac - Toe game #{player1.name} and #{player2.name}"
+
+loop do
+  game.move(player1)
+  break if game.who_won != ''
+
+  puts ''
+  puts '- - - - - - - -'
+
+  # otherwise it will say "It is draw" after leaving the loop
+
+  game.move(player2)
+  break if game.who_won != ''
+end
```

Comment:
## Which turn is 

-  [REQUIRED] game tells which player turn it is.
  - [ ]  How do you which turn is?. Add something an odd and even, where odd is for player 1 turn and even is for player 2 turn

Reply:
Thank you for your review. For each iteration in the loop, both of the players will have a chance to play in a queue. Firstly, the player1 does its move, then player2. It will continue until ```break``` command will be called. The ```break``` command will be called when there is a winner in the game (so ```who_won``` method of ```game``` will tell us that.) 

We haven't planned to use odd or even to switch between players. We have also implemented game logic in Milestone 3. It is working like a charge in there. You can check it in [game_logic](https://github.com/RaminMammadzada/ruby-tic-tac-toe-game/tree/game_logic) branch of our project.